agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -137,7 +140,7 @@ CompletableResultCode internalSendByInstrumentationKey(
       byteBuffers = encode(telemetryItems);
       encodeBatchOperationLogger.recordSuccess();
     } catch (Throwable t) {
-      encodeBatchOperationLogger.recordFailure(t.getMessage(), t);
+      encodeBatchOperationLogger.recordFailure(t.getMessage(), t, TELEMETRY_INTERNAL_SEND_ERROR);
```

Comment:
this one i added it just because of the new method signature and it requires msgId.. i can pass null in this case.

Reply:
gzip might throw ioexception, i think it's ok to keep it. so that it's not just for throttle but encoding error as well? it's hard to check the message since it's not custom.

Reply:
I don't think gzip will throw ioexception since we're zipping in-memory